### ISSUE TITLE
Training on NTU 60 XSUB

### ISSUE THREAD

**Issue by erinchen824** [hehefan/Point-Spatio-Temporal-Convolution#10]
Thanks for your awesome work and generous code sharing.
Recently, I have tried to reproduce the results on NTU 60 XSUB benchmark with one 3090 GPU.
However, it seems the training is really slow and it would take more than a week to train 20 epochs following the original setting.
I have tried to enlarge the batch size, and it still quite slow.
So, what is the reasonable training time for NTU 60 dataset?
Is there anything I have missed?

Looking forward to your reply.

Thanks

**Comment by erinchen824** [comment]
> Thanks for your awesome work and generous code sharing. Recently, I have tried to reproduce the results on NTU 60 XSUB benchmark with one 3090 GPU. However, it seems the training is really slow and it would take more than a week to train 20 epochs following the original setting. I have tried to enlarge the batch size, and it still quite slow. So, what is the reasonable training time for NTU 60 dataset? Is there anything I have missed?
> 
> Looking forward to your reply.
> 
> Thanks

I modify the dataloader to only sample one sequence for each video, and here are part of the training logs.

![截屏_-_5月10日_下午9_02](https://user-images.githubusercontent.com/16219221/167635141-5a51bb1d-078b-4d8b-baec-1b4c112ca2e6.png)

**Comment by hehefan** [comment]
Hi,

I think the training time is reasonable.  One reason is that we used all frames with all combinations, which is a common problem with video processing.   Selecting a fixed frame combination for each video can significantly reduce time but may slightly affect the accuracy.  Another reason is that we did not implement PSTConv in a parallel way like the standard convolutions in torch.nn. Point tubes are sequentially constructed over time.

Best

**Comment by erinchen824** [comment]
> Hi,
> 
> I think the training time is reasonable. One reason is that we used all frames with all combinations, which is a common problem with video processing. Selecting a fixed frame combination for each video can significantly reduce time but may slightly affect the accuracy. Another reason is that we did not implement PSTConv in a parallel way like the standard convolutions in torch.nn. Point tubes are sequentially constructed over time.
> 
> Best

Thanks for your quick reply, that resolves my question.
